.gitignore

```diff
@@ -17,3 +17,9 @@ tmp/
 
 /c.out
 /c.out.tmp
+
+# Documentation
+## Virtualenv
+/.venv
+## MkDocs rendered site
+/site
```

Comment:
`/site` is MkDocs standard build dir, `.venv` is the most common in-project venv directory (poetry, pdm... use `.venv` as default) and so the one used in this project Makefile.
It will be hard not to include them in the `.gitignore` unless you prefer not touching the `.gitignore` at the cost of maintaining lots of settings for each tool (when possible) to change their defaul behavior.

Your `.gitignore` is so small, you shouldn't be afraid to add lines in there. Take a look at [the `.gitignore` from gitignore.io (those use on github, gitlab...) for Python](https://www.toptal.com/developers/gitignore/api/python). I chose to only include those two lines given it is primarily a Go project. But you can't say what tools contributors will use and most of the time, it's good to have an exhaustive `.gitignore` file preventing contributors to commit their tooling generated files (avoid unnecessary CI runs, privacy concern on some included files, time lost while reviewing...)